src/ansys/fluent/core/data_model_cache.py

```diff
@@ -0,0 +1,51 @@
+from collections import defaultdict
```

Comment:
@mkundu1 in this module we'll need a function to recursively update the nested dict state